ProductService/src/main/java/com/metaphorce/exam/ecommerce/product/controller/InventoryController.java

```diff
@@ -0,0 +1,138 @@
+package com.metaphorce.exam.ecommerce.product.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.metaphorce.exam.ecommerce.product.dto.InventoryDTO;
+import com.metaphorce.exam.ecommerce.product.mapper.InventoryMapper;
+import com.metaphorce.exam.ecommerce.product.model.impl.Inventory;
+import com.metaphorce.exam.ecommerce.product.service.impl.InventoryService;
+
+import jakarta.validation.Valid;
+import lombok.extern.slf4j.Slf4j;
+
+@RequestMapping("/inventory")
+@RestController
+@Slf4j
+public class InventoryController {
+	@Autowired
+	private InventoryService service;
+	
+	/**
+	 * Implements the HTTP Post to create a Inventory
+	 * @param Inventory Data to insert
+	 * @return New Inventory create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024 
+	 */
+	@PostMapping
+	public ResponseEntity<InventoryDTO> createInventory(@Valid @RequestBody InventoryDTO InventoryRequest) {
+		log.info("Inventory to create: " + InventoryRequest);
+		ResponseEntity<InventoryDTO> response = null;
+		Inventory newInventory = this.service.create(InventoryMapper.mapper.toEntity(InventoryRequest));
+
```

Comment:
I don't understand why check if the value is null. It would be better to throw exceptions in the service when it doesn't find a certain resource; also, add a global exception handler.

In the case of not being able to create a resource, I think it's better to let a 500 status be generated because it probably means there's something wrong with the system.